### PR TITLE
delete events when receive 404 from events building block

### PR DESCRIPTION
**Pull request by bingzhang**
## Description
_Please provide a summary of the pull request and the issue it fixes. Please add necessary details, context, dependencies, explanation of when review is needed (see next section), etc._

**Fixes #(add issue number here and remove parentheses)**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [ ] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [ ] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

File: utilities/source_utilities.py [hunk 1]
```diff
@@ -411,7 +411,7 @@ def delete_events_in_building_block(objectId_list_to_delete):
         event = find_one(current_app.config['EVENT_COLLECTION'], condition= _id)
         url = current_app.config['EVENT_BUILDING_BLOCK_URL'] + '/' + str(event.get('platformEventId'))
         result = requests.delete(url, headers=headers)
-        if result.status_code != 202:
+        if result.status_code != 202 or result.status_code != 404:
```
Comment (sandeep-ps) [inline utilities/source_utilities.py:414]: @bingzhang, I think this needs to be an AND operation as only if these two statuses are seen, we will be deleting events from the Events Manager local database. Please check. Thanks.

**Comment by sandeep-ps** [comment]
@bingzhang, please fix the merge conflict too. Thanks.

File: utilities/source_utilities.py (hunk 1 above)
Comment (bingzhang) [reply utilities/source_utilities.py:414]: yes you are right.

**Comment by sandeep-ps** [comment]
@bingzhang, could you please fix the merge conflicts too? We can merge this PR after that. Thanks.

File: templates/events/event-edit.html [hunk 2]
```diff
@@ -292,7 +292,7 @@ <h5> Start Date<span style="color: red">*</span> </h5>
               <div class="col-sm-6">
                   {% if post.allDay == True %}
                   <input class="form-control" type="date" id="startDate" name="startDate" value="{{post['startDate']}}" max="9999-12-31" style="width:170px; float:left" required/>
-                  <input class="form-control" type="time" id="startTime" name="startTime" value="00:00" style="width:170px" hidden/>
+                  <input class="form-control" type="time" id="startTime" name="startTime" value="00:00" style="width:170px" hidden/>f
```
Comment (sandeep-ps) [inline templates/events/event-edit.html:295]: There is an extra character here.

**Review by sandeep-ps — APPROVED** [review]

**Pull request merged**